Library/SharedFunctions.swift

```diff
@@ -388,3 +388,94 @@ public func isEndDateAfterToday(for reward: Reward) -> Bool {
   return (reward.endsAt == nil || (reward.endsAt ?? 0) >= AppEnvironment.current.dateType.init()
     .timeIntervalSince1970)
 }
+
+/*
+ A helper that assists in rounding an amount to a given number of decimal places
+ */
+public func rounded(_ value: Double, places: Int) -> Double {
+  let divisor = pow(10.0, Double(places))
+  return (value * divisor).rounded() / divisor
+}
+
+public func checkoutProperties(
+  from project: Project,
+  baseReward: Reward,
+  rewards: [Reward],
+  selectedQuantities: SelectedRewardQuantities,
+  additionalPledgeAmount: Double,
+  pledgeTotal: Double,
+  shippingTotal: Double,
```

Comment:
Same here? If possible, we should try to make the function signature as simple as functionally possible. That way, whenever we need to use it we only need to pass what's absolutely necessary and all the logic required to create `CheckoutPropertiesData` is encapsulated in the method body.